tdx-attestation-app/main.cpp

```diff
@@ -0,0 +1,220 @@
+  //  for an ambiguous match or an extraneous parameter.
+  if (
+      (argv == nullptr) ||
+      (optind >= argc) ||
+      (argv[optind][0] != '-') ||
+      (argv[optind][0] == 0)) {
+    return -1;
+  }
+
+  int opt = argv[optind][1];
+  const char *p = strchr(optstring, opt);
+
+  if (p == NULL) {
+    return '?';
+  }
+  if (p[1] == ':') {
+    optind++;
+    if (optind >= argc) {
+      return '?';
+    }
+    optarg = argv[optind];
+    optind++;
+  }
+  return opt;
+}
+#endif //! PLATFORM_UNIX
+
+void usage(char *programName) {
+  printf("Usage: %s [arguments]\n", programName);
+  printf("   Options:\n");
+  printf("\t-c Config file\n");
+  printf("\t-h Print this help menu\n\n");
+  printf("   Examples:\n");
+  printf("\t%s -c config.json\n", programName);
+  printf("\t%s -h\n", programName);
+}
+
+int main(int argc, char *argv[]) {
+  std::string config_filename;
+  std::string claims_filename;
+
+  int opt;
+  while ((opt = getopt(argc, argv, "c:h")) != -1) {
+    switch (opt) {
+    case 'c':
+      config_filename.assign(optarg);
+      break;
+    case 'h':
+      usage(argv[0]);
+      exit(0);
+    case ':':
+      fprintf(stderr, "Option needs a value\n");
+      exit(1);
+    default:
+      usage(argv[0]);
+      exit(1);
+    }
+  }
+
+  try {
+    // user must provide a config file
+    if (config_filename.empty()) {
+      fprintf(stderr, "Config file is missing\n");
+      usage(argv[0]);
+      exit(1);
+    }
+
+    // set attestation request based on config file
+    std::ifstream config_file(config_filename);
+    json config = json::parse(config_file);
+    config_file.close();
+
+    std::string attestation_url;
+    if (!config.contains("attestation_url")) {
+      fprintf(stderr, "attestation_url is missing\n\n");
+      usage(argv[0]);
+      exit(1);
+    }
+    attestation_url = config["attestation_url"];
+
+    std::string api_key;
+    if (config.contains("api_key")) {
+      api_key = config["api_key"];
+    }
+
+    std::string attestation_type;
+    if (!config.contains("attestation_type")) {
+      fprintf(stderr, "attestation_type is missing\n\n");
+      usage(argv[0]);
+      exit(1);
+    }
+    attestation_type = config["attestation_type"];
+
+    if (!Utils::case_insensitive_compare(attestation_type, "amber") &&
+        !Utils::case_insensitive_compare(attestation_type, "maa")) {
+      fprintf(stderr, "Attestation type was incorrect\n\n");
+      usage(argv[0]);
+      exit(1);
+    }
+
+    // check for user claims
+    std::string client_payload;
+    json user_claims = config["claims"];
+    if (!user_claims.is_null()) {
+      client_payload = user_claims.dump();
+    }
+
+    // if attesting with Amber, we need to make sure an API token was provided
+    // or it was added as an environment variable
+    if (api_key.empty() && Utils::case_insensitive_compare(attestation_type, "amber")) {
+      const char *api_key_value = std::getenv(AMBER_API_KEY_NAME);
```

Comment:
why is this coming from the environment variable? Can we put this in the configuration file as well?

Reply:
this is because Amer currently uses the environment variable as default. But will remove it since this needs to be provided in the config now